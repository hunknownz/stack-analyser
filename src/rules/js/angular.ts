import { register } from '../../rules.js';

register({
  tech: 'angular',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
