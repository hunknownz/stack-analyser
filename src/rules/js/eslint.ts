import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: [
    '.eslintrc',
    '.eslintrc.cjs',
    '.eslintrc.json',
    '.eslintrc.js',
    'eslint.config.js',
    'eslint.config.cjs',
    'eslint.config.mjs',
  ],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
