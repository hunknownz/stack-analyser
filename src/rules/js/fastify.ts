import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'fastify' }],
});
