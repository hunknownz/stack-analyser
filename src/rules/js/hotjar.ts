import { register } from '../../register.js';

register({
  tech: 'hotjar',
  name: 'Hotjar',
  type: 'tool',
  dependencies: [{ type: 'npm', name: '@hotjar/browser' }],
});
