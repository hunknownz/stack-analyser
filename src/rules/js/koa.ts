import { register } from '../../register.js';

register({
  tech: 'koa',
  name: 'Koa',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'koa' }],
});
