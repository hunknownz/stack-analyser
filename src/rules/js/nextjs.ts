import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'framework',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
