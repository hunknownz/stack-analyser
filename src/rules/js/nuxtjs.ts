import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'framework',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
