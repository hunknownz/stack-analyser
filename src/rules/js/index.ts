import './angular.js';
import './bootstrap.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './hotjar.js';
import './koa.js';
import './prettier.js';
import './prisma.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
