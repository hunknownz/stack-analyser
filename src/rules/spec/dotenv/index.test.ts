import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';

import '../../../autoload.js';

describe('dotenv', () => {
  it('should match all dependencies', async () => {
    const env: string[] = [];
    for (const item of rawList) {
      if (item.type !== 'dotenv' || !('dotenv' in item.ref)) {
        continue;
      }

      env.push(`${item.ref.dotenv.map((v) => `${v}KEY=example`).join('\r\n')}`);
    }

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['.env.example'],
        },
        files: {
          '/.env.example': env.join(''),
        },
      }),
    });

    const merged = flatten(res, { merge: true });

    expect([...merged.techs].sort()).toMatchSnapshot();
    expect([...merged.dependencies].sort()).toMatchSnapshot();
  });
});
