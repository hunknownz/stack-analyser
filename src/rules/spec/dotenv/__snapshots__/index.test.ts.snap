// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dotenv > should match all dependencies 1`] = `
[
  "adyen",
  "aircall",
  "airtable",
  "akamai",
  "algolia",
  "amplitude",
  "anthropic",
  "apify",
  "apple",
  "asana",
  "auth0",
  "aws",
  "aws.cloudfront",
  "azure",
  "azure.maps",
  "beehiiv",
  "betterauth",
  "betterstack",
  "blackfire",
  "brevo",
  "bugsnag",
  "calendly",
  "chargebee",
  "clerk",
  "clicksend",
  "clickup",
  "cloudbees",
  "cloudflare",
  "cohereai",
  "contentful",
  "convexdb",
  "databricks",
  "datadog",
  "deepseek",
  "discord",
  "docusign",
  "dokku",
  "dropbox",
  "dynatrace",
  "elasticsearch",
  "facebook",
  "fastly",
  "figma",
  "firebase",
  "firecrawl",
  "freshdesk",
  "gcp",
  "gcp.maps",
  "gcp.vertex",
  "geminiai",
  "ghost",
  "google",
  "google.analytics",
  "groq",
  "heremaps",
  "hotjar",
  "huggingface",
  "ifttt",
  "infisical",
  "inngest",
  "instagram",
  "intercom",
  "intuit",
  "kibana",
  "klarna",
  "lago",
  "launchdarkly",
  "lemonsqueezy",
  "linear",
  "linkedin",
  "logsnag",
  "logtoio",
  "mailchimp",
  "mailjet",
  "mapbox",
  "matomo",
  "meilisearch",
  "memcached",
  "milvusdb",
  "miro",
  "mistralai",
  "mixpanel",
  "mongodb",
  "mysql",
  "n8n",
  "nango",
  "newrelic",
  "notion",
  "okta",
  "ollama",
  "openai",
  "opentelemetry",
  "optimizely",
  "ovh",
  "paddle",
  "pagerduty",
  "papertrail",
  "paypal",
  "perplexityai",
  "pinecone",
  "pingdom",
  "plausible",
  "postgresql",
  "posthog",
  "qdrant",
  "rabbitmq",
  "railway",
  "reddit",
  "redis",
  "resend",
  "rollbar",
  "salesforce",
  "segment",
  "sentry",
  "signoz",
  "slack",
  "snyk",
  "splunk",
  "sqreen",
  "squareup",
  "strapi",
  "stripe",
  "supabase",
  "telegram",
  "tinybird",
  "tursodb",
  "twilio",
  "twitch",
  "twitter",
  "upstash",
  "upstash.kafka",
  "upstash.redis",
  "valkey",
  "webflow",
  "wiz",
  "woocommerce",
  "wordpress",
  "workos",
  "xai",
  "youtube",
  "zapier",
  "zendesk",
  "zoom",
  "zuora",
]
`;

exports[`dotenv > should match all dependencies 2`] = `[]`;
